src/Gizmos/axisScaleGizmo.ts

```diff
@@ -139,34 +159,81 @@ export class AxisScaleGizmo extends Gizmo {
                 this._matrixChanged();
             }
         });
+        // On Drag Listener: to move gizmo mesh with user action
+        this.dragBehavior.onDragObservable.add((e) => increaseGizmoMesh(e.dragDistance));
+        this.dragBehavior.onDragEndObservable.add(resetGizmoMesh);
+
+        // Listeners for Universal Scalar
+        document.addEventListener('universalGizmoDrag', (e) => increaseGizmoMesh((e as any).detail));

```

Comment:
would it be possible to use observables intead of event registration here? especially since you are the one creating and triggering them? Reason is that with observables we are sure when they are executed (inside the render loop)